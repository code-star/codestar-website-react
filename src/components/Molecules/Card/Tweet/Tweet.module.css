.tweet {
	display: flex;
	flex-direction: column;
	border-style: solid;
	border-color: #c6c6c6;
	border-width: 1px;
	margin: 3px;
}
.tweetHeader {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	padding: 5px;
}
.tweetHeader svg {
	cursor: pointer;
}
.tweetBody {
	padding: 5px;
}
.tweetFooter {
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	padding: 5px;
}
.tweetFooter svg {
	cursor: pointer;
}
