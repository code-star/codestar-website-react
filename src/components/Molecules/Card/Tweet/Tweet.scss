@import '../../../../theming/config.scss';

.tweet {
	display: flex;
	flex-direction: column;

	border-style: solid;
	border-color: darken($color-event-blue, 10);
	border-width: 1px;
	border-radius: 5px;

	margin: 5px;
	background-color: $color-white;

	&Header {
		display: flex;
		flex-direction: row;
		justify-content: space-between;
		padding: 5px 5px 0 5px;

    svg {
      cursor: pointer;
    }
	}

	&Body {
		padding: 5px;
	}

	&Footer {
		display: flex;
		flex-direction: row;
		justify-content: space-between;
		padding: 0 5px 5px 5px;

    svg {
      cursor: pointer;
    }
	}
}
