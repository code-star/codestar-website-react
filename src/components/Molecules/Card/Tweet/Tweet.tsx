import React from 'react';
import classNames from 'classnames/bind';
import style from './Tweet.module.css';
import Label from '../../../Atoms/Text/Label';
import Icon from '../../../Atoms/Image/Icon';
import { EColor, ESize, EIcon } from '../../../../utility/enumeration';

const cx = classNames.bind(style);

interface ITweetProps {
	name: string;
	screenName: string;
	createdAt: number;
	text: string;
	favoriteCount: number;
	retweetCount: number;
}

const Tweet = ({
	name,
	screenName,
	createdAt,
	text,
	favoriteCount,
	retweetCount,
}: ITweetProps) => (
	<div className={cx('tweet')}>
		<div className={cx('tweetHeader')}>
			<Label color={EColor.black} size={ESize.medium} text={name} />
			<Label color={EColor.grey} size={ESize.medium} text={`@${screenName}`} />
			<Label
				color={EColor.grey}
				size={ESize.medium}
				text={new Date(createdAt).toDateString()}
			/>
			<Icon name={EIcon.cheveronDown} width={20} height={20} />
		</div>
		<div className={cx('tweetBody')}>
			<Label color={EColor.grey} size={ESize.small} text={text} />
		</div>
		<div className={cx('tweetFooter')}>
			<Icon name={EIcon.bubble} width={18} height={18} fill="#ababab" />
			<Icon
				name={EIcon.swap}
				width={18}
				height={18}
				fill="#ababab"
				text={String(retweetCount)}
			/>
			<Icon
				name={EIcon.heart}
				width={18}
				height={18}
				fill="#ababab"
				text={String(favoriteCount)}
			/>
			<Icon name={EIcon.mail} width={18} height={18} fill="#ababab" />
		</div>
	</div>
);

export default Tweet;
