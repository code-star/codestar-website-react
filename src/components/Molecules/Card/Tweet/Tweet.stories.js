import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, number } from '@storybook/addon-knobs';
import Tweet from './Tweet.tsx';
import { tweets } from '../../../../data/events/tweets';

const {
	created_at,
	favorite_count,
	retweet_count,
	user: { name, screen_name },
} = tweets[3];

storiesOf('Components/Molecules/Card', module)
	.addDecorator(withKnobs)
	.add('Tweet', () => {
		const knobs = {
			name: text('Name', name),
			screenName: text('Screen Name', screen_name),
			createdAt: text('Created At', created_at),
			text: text('Text', tweets[3].text),
			favoriteCount: number('Favorite Count', favorite_count),
			retweetCount: number('Retweet Count', retweet_count),
		};

		return (
			<Tweet
				name={knobs.name}
				screenName={knobs.screenName}
				createdAt={knobs.createdAt}
				text={knobs.text}
				favoriteCount={knobs.favoriteCount}
				retweetCount={knobs.retweetCount}
			/>
		);
	});
