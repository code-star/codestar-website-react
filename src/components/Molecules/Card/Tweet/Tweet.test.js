import React from 'react';
import { shallow } from 'enzyme';

import Tweet from './Tweet';

const getComponent = ({
  name,
  screenName,
  createdAt,
  text,
  favoriteCount,
  retweetCount,
}) => (
  <Tweet
    name={name}
    screenName={screenName}
    createdAt={createdAt}
    text={text}
    favoriteCount={favoriteCount}
    retweetCount={retweetCount}
  />
);

const someTweet = {
  name: 'someAwesomeName',
  screenName: 'someAwesomeScreenName',
  createdAt: 'Fri Aug 31 13:51:31 +0000 2018',
  text: 'some awesome text',
  favoriteCount: 3,
  retweetCount: 5,
};

const renderShallow = () => {
  return shallow(getComponent(someTweet));
};

describe('<Tweet />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of Tweet', () => {
      wrapper = renderShallow();
      expect(wrapper.find('Tweet')).toBeTruthy();
    });
  });

  describe('Snaphot', () => {
    test('must match some tweet', () => {
      expect(global.renderToJSON(getComponent(someTweet))).toMatchSnapshot();
    });
  });
});
