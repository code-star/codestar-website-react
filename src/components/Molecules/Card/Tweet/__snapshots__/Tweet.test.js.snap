// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tweet /> Snaphot must match some tweet 1`] = `
<div
  className="tweet"
>
  <div
    className="tweetHeader"
  >
    <span
      className="label labelBlack labelMedium"
    >
      someAwesomeName
    </span>
    <span
      className="label labelBlack labelMedium"
    >
      @someAwesomeScreenName
    </span>
    <span
      className="label labelBlack labelMedium"
    >
      Fri Aug 31 2018
    </span>
  </div>
  <div
    className="tweetBody"
  >
    <span
      className="label labelBlack labelSmall"
    >
      some awesome text
    </span>
  </div>
</div>
`;
